### ISSUE TITLE
fullscreen mode with hide the system bars (after v2.15.5)

### ISSUE THREAD

**Issue by cygnetlakes** [ankidroid/Anki-Android#9267]
hi, there.
after updated to v2.15.5, fullscreen mode with hide the system bars, there are spaces at the top and the bottom, not fully hide the system bars, and v2.15.4 works perfectly before.
many thanks.
![Screenshot_01](https://user-images.githubusercontent.com/87444056/125690304-4b0b4363-f9ba-40c3-927f-a66502d587d2.png)
![Screenshot_02](https://user-images.githubusercontent.com/87444056/125690302-c1cfd243-759f-4ada-8344-2804df572705.png)
![Screenshot_03](https://user-images.githubusercontent.com/87444056/125690303-375b8f2a-2285-4888-ad01-7571de691b1c.png)

**Comment by prateek-singh-3212** [comment]
Thanks @cygnetlakes for reporting this issue. I am unable to reproduce this issue in my device (AnkiDroid Version 2.15.6). Please share the debug info so that we can investigate further.

For Debug info GOTO:- Settings -> Advanced -> About AnkiDroid

**Comment by AbdelrahmanHamdy1998** [comment]
Yes Can confirm that's also happening with me

AnkiDroid Version = 2.15.6

Android Version = 11

Manufacturer = samsung

Model = SM-G970F

Hardware = exynos9820

Webview User Agent = Mozilla/5.0 (Linux; Android 11; SM-G970F Build/RP1A.200720.012; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/91.0.4472.120 Mobile Safari/537.36

ACRA UUID = 25d26066-b9fd-4cb1-a98f-c863690352d3

Scheduler = std2

Crash Reports Enabled = true

DatabaseV2 Enabled = true

**Comment by david-allison-1** [comment]
```
e323c5291f9ca865718176ab56cd30f22ff71662: ok
b752c5e26c6d09823eccedd95647ae284d5b609d: bad
```

Cause: b752c5e26c6d09823eccedd95647ae284d5b609d

**Comment by david-allison-1** [comment]
Cause: https://github.com/ankidroid/Anki-Android/blob/b752c5e26c6d09823eccedd95647ae284d5b609d/AnkiDroid/src/main/res/layout/navigation_drawer_layout.xml#L7

@mikehardy @ShridharGoel

**Comment by mikehardy** [comment]
The actual commit if you unwind the chain of cherry picks and reverts is https://github.com/ankidroid/Anki-Android/commit/4e914eae50aee209a1e416671c3463eee766700f

Release branch is in sync now with main branch though so should be reproducible on main branch as well, fixed there then cherry picked for another release depending on demand and 2.16 timing

**Comment by cygnetlakes** [comment]
thank you all very much for the help

##
have tested on some devices (android version 5.11/7.11/11.0, and ankidroid version 2.15.4/2.15.5/2.15.6/2.16a5)
there are the same issues for [hide the system bars and the answer buttons]

here is the one of debug info:
AnkiDroid Version = 2.16alpha5

Android Version = 7.1.1

Manufacturer = motorola

Model = Nexus 6

Hardware = shamu

Webview User Agent = Mozilla/5.0 (Linux; Android 7.1.1; Nexus 6 Build/N8I11F; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/85.0.4183.101 Mobile Safari/537.36

ACRA UUID = 1788df22-c33b-4ec9-9d43-e3dccfe86d41

Scheduler = std2

Crash Reports Enabled = true

DatabaseV2 Enabled = true

##
besides, there is a little issue of layout at top-right corner, if using tablet (after v2.15.5)
best regards!
![Screenshot_01](https://user-images.githubusercontent.com/87444056/125849680-1e114b06-81a8-4e5c-ac78-689de542f03b.png)
![Screenshot_02](https://user-images.githubusercontent.com/87444056/125849690-bc0154bb-50ef-47ef-b9ec-2c0f6baa843c.png)

debug info:
AnkiDroid Version = 2.15.6

Android Version = 7.1.1

Manufacturer = htc

Model = Nexus 9

Hardware = flounder

Webview User Agent = Mozilla/5.0 (Linux; Android 7.1.1; Nexus 9 Build/N4F27P; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/53.0.2785.135 Safari/537.36

ACRA UUID = adfe1c1d-c34d-4c2e-823f-8ec742d3dc77

Scheduler = std2

Crash Reports Enabled = true

DatabaseV2 Enabled = true

**Comment by david-allison-1** [comment]
Stared at this for a few hours, turned out that my original statement was correct.

`android:fitsSystemWindows="true"` is set in most views, but not in `reviewer_fullscreen`: 

(2.14 below)

https://github.com/ankidroid/Anki-Android/blob/8732fe311fc3a6813b0ad7b4d1a6b4e8b5ebd655/AnkiDroid/src/main/res/layout/reviewer_fullscreen.xml#L1-L5

We need a way to optionally set this

**Comment by Anthropos888** [comment]
Hi, will this fix be available as update for the stable version soon or should I install the alpha?

**Comment by david-allison-1** [comment]
@mikehardy

**Comment by mikehardy** [comment]
I feel like I've been spamming the release channel and takes my time to do, which is pretty constrained at the moment, so I'm hoping to avoid another 2.15 and instead get a 2.16 out, which is to say there won't be a 2.15 release *at least* for a week or two, but ideally I won't do a 2.15.7 unless this is driving everyone batty. A reasonable compromise might be to issue a 2.15.7beta1 which is much less intensive then you just do beta instead of going all the way to alpha. Let me do that - it'll be out later today

**Comment by mikehardy** [comment]
Never mind - unfortunately @Anthropos888 this fix is no longer cherry-pickable to the release branch, it is involved in a tangle of commits that refactored the affected area and I can't get it in.

If you want the fix for this before an eventual 2.16 release, it will only be accessible via alphas, apologies
